types/promise-fn-retry/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for promise-fn-retry 1.0
+// Project: https://github.com/felippemauricio/promise-fn-retry
+// Definitions by: Jonathan Chang <https://github.com/Jono-Chang>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface RetryOptions {
+  times?: number;
+  initialDelayTime?: number;
+  onRetry?: (err: unknown, optionsParsed: RetryControlOptions) => void;
+  shouldRetry?: (err: unknown) => boolean;
+}
+
+export interface RetryControlOptions extends RetryOptions {
+  retained: number;
+  currentDelay?: number;
+}
```

Comment:
aren't those internal? (so won't be usable really):
https://github.com/felippemauricio/promise-fn-retry/blob/a9f50af2cfd6f58ba43c5431fdcb362132228ad0/src/index.js#L12

Reply:
I do believe it was intentionally exposed on line 9 ('onRetry?: (err: unknown, optionsParsed: RetryControlOptions) => void;') to give the user easy read access to the state. If you think otherwise, I'm more than happy to remove that.